statistics/selectivity.go

```diff
@@ -283,6 +283,66 @@ func (coll *HistColl) Selectivity(ctx sessionctx.Context, exprs []expression.Exp
 			ret *= selectionFactor
 		}
 	}
+
+	// Now we try to cover those still not covered DNF conditions using independence assumption.
+	if mask > 0 {
```

Comment:
How about we use a simpler approach, i.e, make `Selectivity()` a recursive procedure? for example:
```golang
func Selectivity() {
    // the greedy algorithm
    ...
    //
    if mask > 0 {
        dnfItems := FlattenDNFConditions()
        if len(dnfItems) > 0 {
            partSels := make([]float64, len(dnfItems))
            for i, dnfItem := range dnfItems {
                cnfItems := FlattenCNFConditions(dnfItem)
                partSels[i] = Selectivity(cnfItems)
            }
        }
        // compute the selectivity for the DNF filter using partSels.
        ...
    }
}
```

Reply:
We need to make sure each dnfItem only contains a single column. And we also need to union ranges of dnfItems containing the same column, i.e. `a > 3 or b = 4 or a > 6` => `a:(6, +inf] b:[4,4]`. So I think we can't use `Selectivity()` directly.

Reply:
Since we are using independent assumption among the DNF items, why must each DNF item contain only a single column? For the second problem, can we preprocess the filters to simplify the cases you mentioned to `a > 3 or b = 4` first, and then a recursive `Selectivity()` is feasible?